source/include/signaling_data_types.h

```diff
@@ -0,0 +1,381 @@
+
+/* Standard includes. */
+#include <stdint.h>
+#include <stddef.h>
+
+/*-----------------------------------------------------------*/
+
+/**
+ * Maximum number of URIs for an ICE server.
+ */
+#define SIGNALING_ICE_SERVER_MAX_URIS ( 4 )
+
+/**
+ * Maximum length of a channel name.
+ */
+#define SIGNALING_CHANNEL_NAME_MAX_LEN ( 256 )
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Constants for signaling channel TTL.
+ *        Refer https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_SingleMasterConfiguration.html#KinesisVideo-Type-SingleMasterConfiguration-MessageTtlSeconds for details.
+ */
+#define SIGNALING_CHANNEL_TTL_SECONDS_BUFFER_MAX    ( 5 ) /* Maximum length of buffer needed to store TTL. */
+#define SIGNALING_CHANNEL_TTL_SECONDS_MIN           ( 5 )
+#define SIGNALING_CHANNEL_TTL_SECONDS_MAX           ( 120 )
+
+/**
+ * @brief Constants for ICE server TTL.
+ *        Refer https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_signaling_IceServer.html for details.
+ */
+#define SIGNALING_ICE_SERVER_TTL_SECONDS_BUFFER_MAX ( 7 ) /* Maximum length of buffer needed to store TTL. */
+#define SIGNALING_ICE_SERVER_TTL_SECONDS_MIN        ( 30 )
+#define SIGNALING_ICE_SERVER_TTL_SECONDS_MAX        ( 86400 )
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Return code of Signaling component.
+ */
+typedef enum SignalingResult
+{
+    SIGNALING_RESULT_OK,
+    SIGNALING_RESULT_BAD_PARAM,
+    SIGNALING_RESULT_SNPRINTF_ERROR,
+    SIGNALING_RESULT_OUT_OF_MEMORY,
+    SIGNALING_RESULT_INVALID_JSON,
+    SIGNALING_RESULT_UNEXPECTED_RESPONSE,
+    SIGNALING_RESULT_INVALID_TTL,
+    SIGNALING_RESULT_INVALID_ENDPOINT,
+    SIGNALING_RESULT_INVALID_PROTOCOL,
+    SIGNALING_RESULT_INVALID_CHANNEL_NAME,
+    SIGNALING_RESULT_INVALID_CHANNEL_TYPE,
+    SIGNALING_RESULT_INVALID_ICE_SERVER_COUNT,
+    SIGNALING_RESULT_INVALID_ICE_SERVER_URIS_COUNT,
+    SIGNALING_RESULT_INVALID_STATUS_RESPONSE,
+    SIGNALING_RESULT_REGION_LENGTH_TOO_LARGE,
+} SignalingResult_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Protocol type of endpoints while parsing get signaling channel endpoint response.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_GetSignalingChannelEndpoint.html for more detail.
+ */
+typedef enum SignalingProtocol
+{
+    SIGNALING_PROTOCOL_NONE = 0,
+    SIGNALING_PROTOCOL_WEBSOCKET_SECURE = 1,
+    SIGNALING_PROTOCOL_HTTPS = 2,
+    SIGNALING_PROTOCOL_WEBRTC = 4,
+    SIGNALING_PROTOCOL_MAX = 0xFF,
+} SignalingProtocol_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief Type of signaling channel.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_ChannelInfo.html#KinesisVideo-Type-ChannelInfo-ChannelType for more detail.
+ */
+/* We cannot name it SignalingChannelType_t in order to avoid naming conflict
+ * with a data type in the existing KVS SDK. */

```

Comment:
signaling channel type?

Reply:
This comment is not clear to me. Would you please elaborate.

Reply:
The definitions here are conflicted with [here in existing KVS SDK](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/4f1d3add4c0f88212fe6646898e5aba140ecbb3c/src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h#L925-L928).